pennylane/qnodes/base.py

```diff
@@ -264,6 +266,55 @@ def qfunc(a, w):
             "The QNode can only be drawn after its CircuitGraph has been constructed."
         )
 
+    def set_trainable_args(self, arg_indices):
+        """Store the indices of quantum function positional arguments
+        that support differentiability.
+
+        Args:
+            args (None or Set[int]): Differentiable positional argument indices. A
+                value of ``None`` means that all argument indices are differentiable.
+        """
+        if not self.mutable and self.circuit is not None:
+
+            if self.get_trainable_args() == arg_indices:
+                return
+
+            raise QuantumFunctionError(
+                "The trainability of arguments on immutable QNodes cannot be modified after the first evaluation."
+            )
+
+        if not arg_indices:
+            # the provided arg_indices are an empty set
+            self._trainable_args = None
+            return
+
+        # Perform validation
+        if not self.func.var_pos and max(arg_indices) > self.func.n_pos:
+            # QNode does not allow variable positional arguments (*args), and
+            # the provided index set contains a value larger than the number
+            # of positional arguments.
+            raise ValueError(
+                f"Argument index not available. QNode has at most {self.func.n_pos} arguments."
+            )
+
+        if min(arg_indices) < 0:
+            raise ValueError("Argument indices must be positive integers")
+
+        if not all(isinstance(i, int) for i in arg_indices):
+            raise ValueError("Argument indices must be positive integers.")
```

Comment:
```suggestion
        if not all(isinstance(i, int) and i>=0 for i in arg_indices):
                    raise ValueError("Argument indices must be positive integers.")
```
can do this in a one-liner